sdk/core/ci.yml

```diff
@@ -35,6 +35,11 @@ extends:
   parameters:
     ServiceDirectory: core
     ServiceToTest: '*'
+    # Opt out of spell checking the public API surface because `ServiceToTest`
+    # forces the `CodeChecks.ps1` to check ALL public API surfaces (including
+    # those which have known spelling errors) by specifying `*` to the
+    # -ServiceDirectory parameter.
+    SpellCheckPublicApiSurface: false
```

Comment:
This is unfortunate. Can we use the `ServiceDirectory` instead of `ServiceToTest` for spell checking?

Reply:
Or have the setting outside of ci.yml?

Reply:
We use the core pipeline to validate engineering changes across all the libraries so now spell-check issues can slip in without the core pipeline failing.

Reply:
> We use the core pipeline to validate engineering changes across all the libraries so now spell-check issues can slip in without the core pipeline failing. 

Is your intent to always validate that there are no spelling errors in all libraries' public surface areas at any given time? If so, I can tell you that this fails today and the details of the failures are in the bugs which are linked to this PR. Until all of the spelling errors in all of the public API surfaces are addressed (false positives properly accounted, spelling errors corrected as necessary) core will fail on this step. 

Once the spelling errors are addressed then we should set this value to `true` and we can start a more strict enforcement of spelling of the public API surface area.

Reply:
That's what we do for all other checks (generated code up to date, snippets replaced in all MD files, etc.) core pipeline does it for all the projects.

>Until all of the spelling errors in all of the public API surfaces are addressed (false positives properly accounted, spelling errors corrected as necessary) core will fail on this step.

That's why I was proposing the opt-out not to be in ci.yml but somewhere else where the core pipeline can pick it up.

Reply:
@pakrym where do you think would be a good place to include this? If we put it into the project file we would have either move the spell check into the msbuild context or we would have to call an msbuild command to retrieve the value.

We should start thinking about how much of the codecheck work we can move into the msbuild context so we can try and avoid all the going in and out of msbuild and PS etc.

Reply:
I propose to use `ignorePath` in the main cspell.json for that for now.

Reply:
That doesn't seem unreasonable. @danieljurek what do you think about that option?

Reply:
I think this is great. At the beginning of this effort I was adding any file which had any spelling error to `ignorePaths` but that got to be too cumbersome. @pakrym proposes something which strikes the right balance between opting out services and keeping the `ci.yml` complexity under control. Let's do it.